edge/pkg/metamanager/client/lease.go

```diff
@@ -0,0 +1,74 @@
+package client
+
+import (
+	"fmt"
+	"time"
+
+	coordinationv1 "k8s.io/api/coordination/v1"
+
+	"github.com/kubeedge/beehive/pkg/core/model"
+	"github.com/kubeedge/kubeedge/edge/pkg/common/message"
+	"github.com/kubeedge/kubeedge/edge/pkg/common/modules"
+)
+
+var leaseClientTimeout = 10 * time.Second
+
+// TODO:
+// Find a better way to set timeout of SendSync for lease msg.
+func SetLeaseClientTimeout(timeout time.Duration) {
+	leaseClientTimeout = timeout
+}
+
+type LeaseGetter interface {
+	Lease(namespace string) LeaseInterface
+}
+
+type LeaseInterface interface {
+	Create(*coordinationv1.Lease) (*coordinationv1.Lease, error)
+	Update(*coordinationv1.Lease) error
+	Delete(name string) error
+	Get(name string) (*coordinationv1.Lease, error)
+}
+
+type leases struct {
+	namespace string
+	send      SendInterface
+}
+
+func newLeases(namespace string, s SendInterface) *leases {
+	return &leases{
+		send:      s,
+		namespace: namespace,
+	}
+}
+
+func (l *leases) Create(lease *coordinationv1.Lease) (*coordinationv1.Lease, error) {
+	resource := fmt.Sprintf("%s/%s/%s", l.namespace, model.ResourceTypeLease, lease.Name)
+	leaseMsg := message.BuildMsg(modules.MetaGroup, "", modules.EdgedModuleName, resource, model.InsertOperation, lease)
+	_, err := l.send.SendSync(leaseMsg, true, nil)
```

Comment:
for insert operation, metamanager will send this msg to  edged again, for node lease, I think we do not ned this

Reply:
Also metamanager will send new response msg from this msg to the cloud. I think the create will fail always